### ISSUE TITLE
Feature request : Stallguard test before homing or moving to new tool

### ISSUE THREAD

**Issue by LostFool** [moggieuk/Happy-Hare#91]
Sometimes encoder reads wrong, strings get in way, or I hit macros in wrong order. Is it possible to use stall guard to check if we can home before full move? Something to try and stop future problems from too many failed homing attempts.

**Comment by luckyandy0714** [comment]
I agree with LostFool, homing shouldn't stop at a random point. It has to stop at the 'correct' position. So, I prefer using a physical endstop but would also like to implement stallguard to check if the gate is stuck by filament before the selector moves. However, I don't know if these two mechanisms can coexist.
